src/battle_ai_main.c

```diff
@@ -1048,10 +1052,24 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
         }
     }
 
+    // check things that aren't move effects (recoil)
+    if (gBattleMoves[move].recoil > 0)
+    {
+        if (AI_IsDamagedByRecoil(battlerAtk))
+        {
+            u32 recoilDmg = max(1, aiData->simulatedDmg[battlerAtk][battlerDef][AI_THINKING_STRUCT->movesetIndex] * max(1, gBattleMoves[move].recoil) / 100);
+            if (!ShouldUseRecoilMove(battlerAtk, battlerDef, recoilDmg, AI_THINKING_STRUCT->movesetIndex))
+                ADJUST_SCORE(-10);
+            }
+    }
+
     // check move effects
     switch (moveEffect)
     {
-        case EFFECT_HIT:
+        case EFFECT_HIT: // only applies to Vital Throw
+            if (gBattleMoves[move].priority < 0 && AI_STRIKES_FIRST(battlerAtk, battlerDef, move) && aiData->hpPercents[battlerAtk] < 40)
+                ADJUST_SCORE(-1);    // don't want to move last
+            break;
```

Comment:
Can you use numberOfHits for this and decrease it by -2 instead? So if the number of hits to KO is 1 decrease score.

Reply:
I don't know what you mean by "numberOfHits" and I think we're pushing AI changes that go beyond the scope of this PR. Above is the unaltered calculation for Vital Throw.

Reply:
I think he means `strikeCount`?

Reply:
> I think he means `strikeCount`?

no, I mean `GetNoOfHitsToKOBattler`. I've showed an example.